### PR TITLE
feat(deps): Simplify dependencies, remove canvas/jsdom

### PR DESCRIPTION
**Pull request by OpenGG**
Problem: While digging around [image.onerror](https://github.com/oslllo/potrace/pull/46#issuecomment-1003050173), I realized the whole `getBase64Async`/`base64ToImage`/`getImageData` are not necessary at all: `jimp` already reads out pixel bitmap, no need for `jsdom`, `image` and `canvas`.

So I figured, why keep canvas as dependency? It is native module, installing can be slow and nasty.

Removing `canvas` was not easy, because `oslllo-svg2` and `oslllo-validator` also depend on it. I had to rewrite some code, and introduced `resvg` wasm for png rendering.

This PR could be controversy, major modifications without prior discussion. But the result is good, both install time and size down by more than 50%.

Before:

```bash
me@MacBook-Pro potrace % yarn
yarn install v1.22.17
info No lockfile found.
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 33.13s.
me@MacBook-Pro potrace % du -sh node_modules 
109M    node_modules
```

Now:
```bash
me@MacBook-Pro potrace % yarn
yarn install v1.22.17
info No lockfile found.
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
✨  Done in 9.79s.
me@MacBook-Pro potrace % du -sh node_modules 
 50M    node_modules
```

### CODE REVIEW

File: example/example.js [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+    await initialize(buffer)
+}
+
+let initPromise;
+
+const draw = async (svg) => {
+    initPromise = initPromise || init()
+    await initPromise
+
+    return svg2png(svg)
```
Comment (Ghustavh97) [inline example/example.js]: Add the background option to make the example output visible for dark mode users.

```js
return svg2png(svg, { backgroundColor: 'white', });
```
Current output

image

**Review by Ghustavh97 — CHANGES_REQUESTED** [review]

**Comment by Ghustavh97** [comment]
With the massive changes I thought this was going to break alot of stuff but with the tests I have done, everything upstream on my ends seems to work as intended. 

I hated having to use jsdom from day one because of that canvas issue, but it was all I could find that could get the job done and making it work was more important at the time. I did later on discover https://github.com/fgnass/domino but im not sure it has the same problem or not. 

I might release this as v2.0 just to be on the safe side tho, but yeah svg2png-wasm seems to be a game changer.

**Comment by OpenGG** [comment]
@Ghustavh97 
Massive change yes, but really just how imageData is parsed, from `jimp` or from `canvas`.

If both module work correctly, pixel data coming from either one should be the same.

But yes, releasing v2 would be more appropriate.

**Review by Ghustavh97 — APPROVED** [review]

**Comment by Ghustavh97** [comment]
Thanks for the PR. If you are ever free and interested please feel free to check out https://github.com/oslllo/svg-fixer & https://github.com/oslllo/svg2 since they both are somewhat related to this package.

**Pull request merged**